src/modules/flow/http-client/http-client.c

```diff
@@ -550,33 +619,43 @@ common_post_process(struct sol_flow_node *node, void *data,
 }
 
 static int
-boolean_process_token(struct sol_flow_node *node,
-    struct sol_json_token *key, struct sol_json_token *value)
+boolean_process_token(struct sol_flow_node *node, struct sol_buffer *buf)
 {
-    bool result;
+    bool result, found;
+    enum sol_json_loop_reason reason;
+    struct sol_json_scanner sub_scanner;
+    struct sol_json_token sub_key, sub_value, token;
 
-    if (sol_json_token_get_type(value) == SOL_JSON_TYPE_TRUE)
-        result = true;
-    else if (sol_json_token_get_type(value) == SOL_JSON_TYPE_FALSE)
-        result = false;
-    else
-        return -EINVAL;
+    sol_json_scanner_init(&sub_scanner, buf->data, buf->used);
+    found = false;
+
+    SOL_JSON_SCANNER_OBJECT_LOOP (&sub_scanner, &token, &sub_key, &sub_value, reason) {
+        if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&sub_key, "value")) {
+            if (sol_json_token_get_type(&sub_value) == SOL_JSON_TYPE_TRUE)
+                result = true;
+            else if (sol_json_token_get_type(&sub_value) == SOL_JSON_TYPE_FALSE)
+                result = false;
+            else
+                return -EINVAL;
+            found = true;
+            break;
+        }
+    }
+
+    SOL_EXP_CHECK(!found, -ENOENT);
 
     return sol_flow_send_boolean_packet(node,
         SOL_FLOW_NODE_TYPE_HTTP_CLIENT_BOOLEAN__OUT__OUT, result);
 }
 
 static int
-boolean_process_data(struct sol_flow_node *node,
-    struct sol_http_response *response)
+boolean_process_data(struct sol_flow_node *node, struct sol_buffer *buf)
 {
     bool result;
 
-    SOL_HTTP_RESPONSE_CHECK_API(response, -EINVAL);
-
-    if (!strncasecmp("true", response->content.data, response->content.used))
+    if (!strncasecmp("true", buf->data, buf->used))
         result = true;
-    else if (!strncasecmp("false", response->content.data, response->content.used))
+    else if (!strncasecmp("false", buf->data, buf->used))
```

Comment:
I don't think I will change this function to use a sol_str_slice, because `get_blob_process()` and `get_json_process()`
uses `sol_buffer_steal()` in order to create blob. If I change to sol_str_slice I will possible duplicate memory.